tests/profiling/collector/test_stack.py

```diff
@@ -761,7 +761,4 @@ def _foo():
             _foo()
 
     gc.collect()  # Make sure we don't race with gc when we check frame objects
-    frametype_objs = [obj for obj in gc.get_objects() if isinstance(obj, FrameType)]
-    for obj in frametype_objs:
-        # Ensure that all frames relating to _foo() have been gc'd already
-        assert obj.f_code.co_name != "_foo"
+    assert sum(isinstance(_, FrameType) for _ in gc.get_objects()) == 0
```

Comment:
wouldn't we potentially have frame objects from the current stack here?

what determines when a `FrameType` object is created? is that not the same as the `PyFrame_Object`(sp?) ?

Reply:
Yes, `FrameType` is referencing `PyFrame_Object`. This is created in the stack collector https://github.com/DataDog/dd-trace-py/blob/a01c18f20de2348ed34bde3ac2fe7a1e010a2d38/ddtrace/profiling/collector/stack.pyx#L248. The fix from #4901 sought to properly decrement its reference counts so that they can be garbage collected instead of leaking strong references.

We do `gc.collect()` the line above to make sure any frame objects that are being tracked are collected before we make this assertion.